Sources/NIOWebSocket/WebSocketFrame.swift

```diff
@@ -0,0 +1,270 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebSocketFrame.swift
+//  NIOWebSocket
+//
+//  Created by Cory Benfield on 16/02/2018.
+//
+
+import NIO
+
+private extension UInt8 {
+    func bitSetWithMask(_ mask: UInt8) -> Bool {
+        return self & mask != 0
+    }
+
+    mutating func setBitForMask(_ mask: UInt8, to: Bool) {
+        if to {
+            self |= mask
+        } else {
+            self &= ~mask
+        }
+    }
+}
+
+/// A single 4-byte websocket masking key.
+///
+/// WebSockets uses a masking key to prevent malicious users from injecting
+/// predictable binary sequences into websocket data streams. This structure provides
+/// a more convenient method of interacting with a masking key than simply by passing
+/// around a four-tuple.
+public struct WebSocketMaskingKey {
+    private let key: (UInt8, UInt8, UInt8, UInt8)
+
+    public init?<T: Collection>(_ buffer: T) where T.Element == UInt8 {
```

Comment:
I think this might just want to take `UInt32` too. Then you can use the `ByteBuffer` methods to read that out instead of making an `Array` that is of no value

Reply:
@Lukasa but here certainly a `UInt32` would make sense? Totally happy for you to unpack then into 4 `UInt8`

Reply:
Sure, that works for me.